torch/utils/_sympy/value_ranges.py

```diff
@@ -171,10 +179,17 @@ def __init__(self, lower: AllIn, upper: AllIn) -> None:
         object.__setattr__(
             self,
             "is_int",
-            not self.is_bool
-            and (
-                isinstance(lower, (sympy.Integer, NegativeIntInfinity))
-                or isinstance(upper, (sympy.Integer, IntInfinity))
+            (
+                not self.is_bool
+                and (
+                    lower == -sympy.oo  # in simple_sympify() we use -oo for integer
```

Comment:
I don't understand, -int_oo should be a valid lower bound here

Reply:
Added `int_oo` for bound as well.